Source/Pawnmorphs/Esoteria/ThingComps/AnimalSelectorComp.cs

```diff
@@ -87,20 +89,38 @@ public IEnumerable<PawnKindDef> AllAnimalsSelectable
         {
             get
             {
-                if (Props.requiresTag)
+                IEnumerable<PawnKindDef> animals = Props.AllAnimals;
+
+                // Handle different modes
+                if (Props.selectionMode == "Tagged")
+                {
+                    // All tagged animals
+                    animals = animals.Where(x => Database.TaggedAnimals.Contains(x));
+                
+                }
+                else if (Props.selectionMode == "NotSequenced")
```

Comment:
I don't like the idea of having a selection mode on a general component that is very sequencer specific. It kind of contradicts the polymorphism and generics.
Instead what you could do is inject the "Not sequenced" filter into the race selector's filter property so that it is the Sequencer that defines how the selector behaves in that specific circumstance.

"SpeciesFilter" callback in this case as "m => !Database.StoredMutations.Contains(m)"

Reply:
Agreed, this should be injected into the SpeciesFilter when the parent & comp is initialized not as a string field that is very sequencer specific

Reply:
The idea was to provide reusable presets for selection in a centralised location, rather than have each caller define (perhaps redundantly) it's own selection mode. It also allows the selector's configuration to remain in the xml and not be scattered around.

Reply:
The problem is that if we had 10 things all using the selector in their own way, then you would need 10 different selection modes, all of them only ever used in a single location.
As opposed to each one defining themselves how it should behave and then injecting it.
Dependency injection even if potentially redundant, will still make it easier to maintain since the related code will be kept together. And no specialised behavior logic would be stored inside a generalised component.